users.js

```diff
@@ -0,0 +1,208 @@
+
+  let idTd = document.createElement("td");
+  let nameTd = document.createElement("td");
+  let emailTd = document.createElement("td");
+  let ageTd = document.createElement("td");
+  let addressTd = document.createElement("td");
+  let editTd = document.createElement("td");
+  let deleteTd = document.createElement("td");
+
+  let editButton = document.createElement("button");
+  let deleteButton = document.createElement("button");
+
+  editButton.innerHTML = "Edit";
+  editButton.className = "button";
+  editButton.addEventListener("click", () => loadUserToForm(user));
+
+  deleteButton.innerHTML = "Delete";
+  deleteButton.className = "button";
+  deleteButton.addEventListener("click", () => deleteUser(user));
+
+  editTd.appendChild(editButton);
+  deleteTd.appendChild(deleteButton);
+
+  idTd.innerHTML = user.id;
+  nameTd.innerHTML = user.name;
+  emailTd.innerHTML = user.email;
+  ageTd.innerHTML = user.age;
+  addressTd.innerHTML = user.address;
+
+  tableRow.appendChild(idTd);
+  tableRow.appendChild(nameTd);
+  tableRow.appendChild(emailTd);
+  tableRow.appendChild(ageTd);
+  tableRow.appendChild(addressTd);
+  tableRow.appendChild(editTd);
+  tableRow.appendChild(deleteTd);
+
+  tableBody.appendChild(tableRow);
+
+  trMap[user.id] = {
+    name: nameTd,
+    email: emailTd,
+    age: ageTd,
+    address: addressTd,
+    row: tableRow,
+  };
+};
+
+const addUserToJSON = (user) => {
+  let database = getDb();
+  let users = database.users;
+  users.push(user);
+  localStorage.setItem("database", JSON.stringify({ ...database, users }));
+};
+
+getUsers().then((usersObj) => {
+  usersObj.forEach(addUserToTable);
+});
+
+let form = document.getElementById("form");
+
+form.addEventListener("submit", (ev) => {
+  ev.preventDefault();
+
+  let id = ev.target.id.value;
+
+  let newId = (Math.random() * 10).toString().replace(".", "").substr(0, 10);
+  let name = ev.target.name.value;
+  let email = ev.target.email.value;
+  let age = ev.target.age.value;
+  let address = ev.target.address.value;
+
+  if (!nameRegex.test(name)) {
+    alert(
+      "Name should be between 1-30 characters, uppercase and lowercase letters and space are allowed."
+    );
+  } else if (!emailRegex.test(email)) {
+    alert("Email should be valid.");
+  } else if (age < 0 || age > 150) {
+    alert("Age should be between 0-150.");
+  } else if (!addressRegex.test(address)) {
+    alert(
+      "Address should be betweeen 1-300 characters and uppercase, lowercase, numbers, space are allowed."
+    );
+  } else {
+    let user = { id: id || newId, name, email, age, address };
+
+    if (id) {
+      editUser(user);
+    } else {
+      addUserToTable(user);
+      addUserToJSON(user);
+    }
+
+    form.reset();
+  }
+});
+
+const loadUserToForm = (user) => {
+  //Loads the data from the local storage, because the form doesn't update until the page is refreshed.
+  let userData = getDb().users.find((u) => u.id === user.id);
+
+  form.id.value = userData.id;
```

Comment:
As long as the form is always present on the page, it'd be good if you focus on it after edit is clicked. Otherwise I need to scroll to top in order to do the changes I wanted.